File: dart/space-age/lib/space_age.dart
class SpaceAge {

  static const SPY = 31557600;

  final Map<String, double> orbitalPeriods = {
    'Earth': 1,
    'Mercury': 0.2408467,
    'Venus': 0.61519726,
    'Mars': 1.8808158,
    'Jupiter': 11.862615,
    'Saturn': 29.447498,
    'Uranus': 84.016846,
    'Neptune': 164.79132
  };


  num age({required String planet, required int seconds}) => num.parse((seconds / SPY / orbitalPeriods[planet]!).toStringAsFixed(2));

}

File: javascript/space-age/space-age.js
const SPY = 31557600;
const MAP = {
    'mercury': 0.2408467,
    'venus': 0.61519726,
    'mars': 1.8808158,
    'jupiter': 11.862615,
    'saturn': 29.447498,
    'uranus': 84.016846,
    'neptune': 164.79132,
    'earth': 1
}

export const age = (planet, sec) => Number((sec / MAP[planet] / SPY).toFixed(2));